sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureFileSystemProvider.java

```diff
@@ -1133,29 +1131,29 @@ void closeFileSystem(String fileSystemName) {
         this.openFileSystems.remove(fileSystemName);
     }
 
-    private String extractAccountName(URI uri) {
+    private String extractAccountEndpoint(URI uri) {
         if (!uri.getScheme().equals(this.getScheme())) {
             throw LoggingUtility.logError(this.logger, new IllegalArgumentException(
                 "URI scheme does not match this provider"));
         }
         if (CoreUtils.isNullOrEmpty(uri.getQuery())) {
             throw LoggingUtility.logError(this.logger, new IllegalArgumentException("URI does not contain a query "
-                + "component. FileSystems require a URI of the format \"azb://?account=<account_name>\"."));
+                + "component. FileSystems require a URI of the format \"azb://?endpoint=<account_endpoint>\"."));
         }
 
-        String accountName = Flux.fromArray(uri.getQuery().split("&"))
-                .filter(s -> s.startsWith(ACCOUNT_QUERY_KEY + "="))
+        String endpoint = Flux.fromArray(uri.getQuery().split("&"))
+                .filter(s -> s.startsWith(ENDPOINT_QUERY_KEY + "="))
                 .switchIfEmpty(Mono.error(LoggingUtility.logError(this.logger, new IllegalArgumentException(
-                        "URI does not contain an \"" + ACCOUNT_QUERY_KEY + "=\" parameter. FileSystems require a URI "
-                            + "of the format \"azb://?account=<account_name>\""))))
-                .map(s -> s.substring(ACCOUNT_QUERY_KEY.length() + 1))
+                        "URI does not contain an \"" + ENDPOINT_QUERY_KEY + "=\" parameter. FileSystems require a URI "
+                            + "of the format \"azb://?endpoint=<endpoint>\""))))
+                .map(s -> s.substring(ENDPOINT_QUERY_KEY.length() + 1)) // Trim the query key and =
```

Comment:
For debug-ability, do you think we could capture all `ENDPOINT_QUERY_KEY`s in the query parameters and if there is more than 1 we log an informational message and indicate that the last value will be selected. That way if someone runs into an issue with our decision they have breadcrumbs to a resolution.